extensions/config-yaml/runtime/src/main/java/io/quarkus/yaml/configuration/runtime/YamlConfigSourceProvider.java

```diff
@@ -0,0 +1,61 @@
+package io.quarkus.yaml.configuration.runtime;
+
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.APPLICATION_YML_FILE;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.APPLICATION_YML_FILESYSTEM_PRIORITY;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.APPLICATION_YML_JAR_PRIORITY;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.MICROPROFILE_CONFIG_YML_FILE;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.MICROPROFILE_CONFIG_YML_FILESYSTEM_PRIORITY;
+import static io.quarkus.yaml.configuration.runtime.YamlConfigConstants.MICROPROFILE_CONFIG_YML_JAR_PRIORITY;
+
+import java.io.UncheckedIOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
+
+public class YamlConfigSourceProvider implements ConfigSourceProvider {
+
+    @Override
+    public Iterable<ConfigSource> getConfigSources(ClassLoader classLoader) {
+        List<ConfigSource> configSources = new ArrayList<>();
+
+        URL jarMicroprofileConfigYml = classLoader.getResource(MICROPROFILE_CONFIG_YML_FILE);
+        if (jarMicroprofileConfigYml != null) {
+            configSources.add(new YamlConfigSource(jarMicroprofileConfigYml, MICROPROFILE_CONFIG_YML_JAR_PRIORITY));
+        }
+        URL jarApplicationYml = classLoader.getResource(APPLICATION_YML_FILE);
+        if (jarApplicationYml != null) {
+            configSources.add(new YamlConfigSource(jarApplicationYml, APPLICATION_YML_JAR_PRIORITY));
+        }
+
+        URL fsMicroprofileConfigYml = getFileSystemURL(MICROPROFILE_CONFIG_YML_FILE);
+        if (fsMicroprofileConfigYml != null) {
+            configSources.add(new YamlConfigSource(fsMicroprofileConfigYml, MICROPROFILE_CONFIG_YML_FILESYSTEM_PRIORITY));
+        }
+        URL fsApplicationYml = getFileSystemURL(APPLICATION_YML_FILE);
+        if (fsApplicationYml != null) {
+            configSources.add(new YamlConfigSource(fsApplicationYml, APPLICATION_YML_FILESYSTEM_PRIORITY));
+        }
+
+        return configSources;
+    }
+
+    private URL getFileSystemURL(String source) {
+        Path path = Paths.get("config", source);
```

Comment:
Not a big deal, but maybe we shouldn't assume that a `config` directory will exist in the same directory as the application

Reply:
Ideally yes, but I think that the `.properties` equivallent does assume this. @dmlloyd had some work in progress to make this configurable on the command line IIRC.

Reply:
Yes, this is also true for the properties-based configuration and documented as well in the [configuration guide](https://github.com/quarkusio/quarkus/blob/master/docs/src/main/asciidoc/application-configuration-guide.adoc):

>using a configuration file placed in $PWD/config/application.properties
>
>By placing an application.properties file inside a directory named config which resides in the directory where the application runs, any runtime properties defined in that file will override the default configuration. Furthermore any runtime properties added to this file that were not part of the original application.properties file will also be taken into account.
>
> This works in the same way for runner jar and the native executable

Reply:
Right, unless the `config` directory is added to the classpath, then the `ClassLoader.getResource` above would grab it